src/providers/local/join-pool.provider.ts

```diff
@@ -284,22 +284,31 @@ export const joinPoolProvider = (
       : tokenApprovalActions;
   }
 
+  // Checks amountsIn for valid inputs and updates price impact state if
+  // invalid.
+  function validateAmountsIn(): boolean {
+    if (!hasAmountsIn.value) {
+      priceImpact.value = 0;
+      return false;
+    }
+
+    return true;
+  }
+
   /**
    * Simulate join transaction to get expected output and calculate price impact.
    */
   async function queryJoin() {
     // If form is empty or inputs are not valid, clear the price impact and
     // return early
-    if (!hasAmountsIn.value) {
-      priceImpact.value = 0;
-      return null;
-    }
+    if (!validateAmountsIn()) return null;
 
     try {
       joinPoolService.setJoinHandler(joinHandlerType.value);
       await setApprovalActions();
 
       console.log('joinHandler:', joinHandlerType.value);
+      if (!validateAmountsIn()) return null;
```

Comment:
So I've added this additional guard right before the query call, it's possible that the amounts were being reset to empty whilst we waited on the `setApprovalActions` function.